### ISSUE TITLE
Display name of chapter in workshop invitation

### ISSUE THREAD

**Issue by brunogirin** [codebar/planner#1372]
## Description of the issue 📄

When receiving a workshop invitation by email, the name of the chapter is not mentioned. The only way to know what chapter it comes from is to check the email address in the contact info at the bottom of the email. Members can be registered to multiple chapters so it would be good to mention the name of the chapter the invitation comes from somewhere in the body of the email.

## Screenshots 📷

![Screenshot_2020-07-29 Gmail - Virtual Workshop Coach Invitation Wed, 29th July 18 30 BST (GMT+01 00)](https://user-images.githubusercontent.com/216028/88778732-c509e300-d180-11ea-9f01-a01d4081de02.png)

## To do 📋

* [ ] Claim this issue (comment below, or assign yourself if you are part of the codebar org)
* [ ] Fork and clone the repository
* [ ] Update the relevant files. Follow the steps to fix section in this issue.
* [ ] Commit your changes as one commit. Use the title of this issue as your commit message
* [ ] Submit a pull request
* [ ] Mention this issue in the PR description by including it's number
* [ ] Have your pull request reviewed & merged by a codebar team member

**Comment by matyikriszta** [comment]
I'll pick this up @brunogirin as this has been nagging me as well.

**Comment by Indrajitnaiya09** [comment]
Hi, I'd like to solve this issue, seems no one has solved it till now. This will be my first contribution to the Open source project. Could you help me a little? @brunogirin 
Can you a little more specific like in email body where I need to display the chapter name.
And in this case, "London" (london@codebar.io) is the chapter name?
Thanks in advance :)

**Comment by despo** [comment]
@Indrajitnaiya09 a summary of things to do

1. Amend invitation title to

> [chapter_name] ...Existing title..

I think this change would be made in the mailers e.g. https://github.com/codebar/planner/blob/master/app/mailers/virtual_workshop_invitation_mailer.rb and https://github.com/codebar/planner/blob/master/app/mailers/workshop_invitation_mailer.rb in `invite_coach` and `invite_student` 

2. I guess it might be worth updating all invitation related mailer titles for consistency. (nice to have)

3. Amend invitation body to include chapter name in first line e.g.
> Invites for the next #{chapter_name} virtual workshop are now open...

To resolve this emails for both workshops and virtual workshops should be updated (can be found [here](https://github.com/codebar/planner/tree/master/app/views/virtual_workshop_invitation_mailer) and [here](https://github.com/codebar/planner/tree/master/app/views/workshop_invitation_mailer)

The chapter's name can be accessed through the workshops model e.g. `workshop.chapter.name`

**Comment by matyikriszta** [comment]
@despo @Indrajitnaiya09 I'm already working on this, I have commented above and added myself to the Assignnees section. Happy to pass it onto you however @Indrajitnaiya09 if you are keen.

**Comment by Indrajitnaiya09** [comment]
> @despo @Indrajitnaiya09 I'm already working on this, I have commented above and added myself to the Assignees section. Happy to pass it onto you however @Indrajitnaiya09 if you are keen.

HI @matyikriszta, Please let me do this. That'll be my first PR. I hope you can understand. Thank you.

**Comment by Indrajitnaiya09** [comment]
@despo could you help me to understand the email trigger thing. I already generated the Mailchimp API key and set-up under the .env file.

Now, I did sign-up as a couch and did register for a chapter. But I'm not getting mail. Is there anything I'm missing!! I think I'm almost done with my code, but want to test before I do PR request. Thank you in advance :)

Note: Application is running in localhost:3000

**Comment by notapatch** [comment]
@Indrajitnaiya09 - the generation of mails is done by a [worker process configured in the Procfile](https://github.com/codebar/planner/blob/master/Procfile) - called delayed job. If you wanted to run the application and worker process you could use something like the gem [foreman](https://github.com/ddollar/foreman). You would then generate emails.

```
± % foreman start                                                                                             !10091
10:47:08 web.1    | started with pid 39503
10:47:08 worker.1 | started with pid 39504              <=== There it is!
10:47:09 web.1    | [39503] Puma starting in cluster mode...
...
10:47:14 web.1    | [39503] - Worker 2 (pid: 39510) booted, phase: 0
10:47:15 worker.1 | [Worker(host:Skye pid:39504)] Starting job worker
10:47:15 worker.1 | D, [2020-09-23T10:47:14.590392 #39504] DEBUG -- :   Delayed::Backend::ActiveRecord::Job Load (75.7ms)  UPDATE "delayed_jobs" SET locked_at = '2020-09-23 10:47:14.469512', locked_by = 'host:Skye pid:39504' WHERE id IN (SELECT  "delayed_jobs"."id" FROM "delayed_jobs" WHERE ((run_at <= '2020-09-23 10:47:14.467429' AND (locked_at IS NULL OR locked_at < '2020-09-23 10:42:14.467483') OR locked_by = 'host:Skye pid:39504') AND failed_at IS NULL)  ORDER BY priority ASC, run_at ASC LIMIT 1 FOR UPDATE) RETURNING *
```

However, you may not need to worry about generating emails if you are only interested in how the email looks? In which case you can see any changes reflected in the mail previewers (basically it generates the email from a model without needing to involve the worker process). 

You can navigate to the mailer previews like so:

`http://localhost:3000/rails/mailers`

![Mailer](https://user-images.githubusercontent.com/1710795/93998483-7fd2ed80-fd8c-11ea-9a29-b3b13e4de4e0.png)

When you click through on a link you see the mail view. For example if I look under `Virtual Workshop Invitation Mailer` and click `invite coach` I will see the email as follows: 

![Screenshot 2020-09-23 at 11 05 05](https://user-images.githubusercontent.com/1710795/93998740-db9d7680-fd8c-11ea-8ff9-a9621a83fd73.png)


If there's any missing or crashing when you click on the link, then that's a bug.

### References
[Rails Guide - Action Mailer Basics - 2.6 Previewing Emails](https://guides.rubyonrails.org/action_mailer_basics.html#previewing-emails)

**Comment by Indrajitnaiya09** [comment]
@notapatch Thanks for helping me. I was not able to get the thing. Now I got some ideas. I'll start working on it. Once again thanks.

**Comment by Indrajitnaiya09** [comment]
Hi @matyikriszta or @notapatch can you assign this issue to me? And @notapatch thanks for helping me.

**Comment by KimberleyCook** [comment]
@Indrajitnaiya09 assigned it to you

**Comment by Indrajitnaiya09** [comment]
@notapatch @KimberleyCook @despo I fixed the issue and did a pr, please do the review.

While submitting the pr, I'm not able to assign anyone. could you please help me with this. Thank you.

**Comment by notapatch** [comment]
@Indrajitnaiya09 - FYI - I am a committer only, I have no role within the project. The active maintainers, in alphabetic order, are: @despo, @KimberleyCook, @matyikriszta.

**Comment by despo** [comment]
Fixed by #1404
